src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/IndexBuilder.java

```diff
@@ -163,9 +163,13 @@ public void addElements()  {
     public void add(IndexItem item) {
         Objects.requireNonNull(item);
 
-        itemsByFirstChar.computeIfAbsent(keyCharacter(item.getLabel()),
+        if (item.isElementItem() || item.isTagItem()) {
+            // don't put summary-page items in the A-Z index:
+            // they are listed separately, at the top of the index page
+            itemsByFirstChar.computeIfAbsent(keyCharacter(item.getLabel()),
```

Comment:
You're preventing these summary pages from showing up in the alphabetic index, but not from the by-category index below. This means they are added to the javadoc search contents, which by itself is a good thing. However, this means we also need to think about the category under which the item is listed. For example, you are using `IndexItem.Category.MEMBER` for the Deprecated API list, but deprecation applies to all element categories. I guess we should use the `TAGS` category for summary pages unless there is a clear element category they fit in?

Reply:
Initially, I put the summary pages in the alphabetic index as well, but it disrupted test files too much. It was sort-of silly to see `All Classes` and `All Packages` showing up under `A` in the alphabetic index when previously there was no `A` entry. Hence the change to the current behavior.

See next comment about which category to use.